torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Dataset splitting utilities."""
+
+from copy import deepcopy
+from math import floor
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+from rtree.index import Index, Property
+from torch import Generator, default_generator, randint, randperm
+from torch._utils import _accumulate
+from torch.utils.data import Subset, TensorDataset, random_split
+
+from ..datasets import GeoDataset, NonGeoDataset
+from .utils import BoundingBox
+
+__all__ = (
```

Comment:
For anything we want to be public, we should add it to `torchgeo/datasets/__init__.py`, everything else is considered to be private.

Reply:
These should also be added to `docs/api/datasets.rst` under the "Utilities" section.

Reply:
Added the new functions to both